Integrations/integration-DuoAdminApi.yml

```diff
@@ -0,0 +1,342 @@
+        demisto.results(entry)
+
+    def get_authentication_logs_by_user(username,mintime):
+        res = admin_api.get_authentication_log(2,
+                                                username=username,
+                                                mintime=time_to_timestamp_milliseconds(OPTIONS_TO_TIME[mintime]),
+                                                maxtime=time_to_timestamp_milliseconds(datetime.datetime.now()),
+                                                limit='50')
+
+        raw_logs = res['authlogs']
+        for log in raw_logs:
+            log['timestamp'] = formatEpochDate(log['timestamp'])
+        entry = get_entry_for_object('Authentication logs for ' + username, raw_logs, raw_logs,
+                {'DuoAdmin.UserDetails(val.username && val.username == obj.username)':{ 'username': username,'auth_logs': raw_logs}},
+        headers = ['access_device','event_type','result','reason','application','factor','timestamp','auth_device'])
+        demisto.results(entry)
+
+    def get_devices_by_user(username):
+        user_id = get_user_id(username)
+        res = admin_api.get_user_phones(user_id)
+        entry = get_entry_for_object('Devices for ' + username, res, res,
+                {'DuoAdmin.UserDetails(val.username && val.username == obj.username)':{ 'username': username,'phones' : res}})
+        demisto.results(entry)
+
+    def get_all_devices():
+        res = admin_api.get_phones()
+        entry = get_entry_for_object('Devices',res,res,
+            {'DuoAdmin.Phones(val.phone_id==obj.phone_id)' : res})
+        demisto.results(entry)
+
+    def dissociate_device_by_user(username, device_id):
+        user_id = get_user_id(username)
+        res = admin_api.delete_user_phone(user_id,device_id)
+        demisto.results('Phone with ID ' + device_id + 'was dissociated from user ' + username)
+
+    def associate_device_to_user(username,device_id):
+        user_id = get_user_id(username)
+        res = admin_api.add_user_phone(user_id ,device_id)
+        demisto.results('Phone with ID ' + device_id + 'was associated to user ' + username)
+
+    def get_u2f_tokens_by_user(username):
+        user_id = get_user_id(username)
+        res = admin_api.get_user_u2ftokens(user_id)
+        for token in res:
+            token['date_added'] = formatEpochDate(token['date_added'])
+        entry = get_entry_for_object('U2F Tokens for ' + username, res, res,
+                {'DuoAdmin.UserDetails(val.username && val.username == obj.username)':{ 'username': username,'u2ftokens' : res}})
+        demisto.results(entry)
+
+    def delete_u2f_token(token_id):
+        res = admin_api.delete_u2ftoken(token_id)
+        demisto.results('Token with ID ' + token_id + ' deleted successfully')
+
+    #Execution
+
+    if demisto.command() == 'test-module':
+        try:
+            res = admin_api.get_info_summary()
+            demisto.results('ok')
+        except:
+            demisto.results('not ok')
```

Comment:
Don't do this. Catch the exception and provide a meaningful error based on the caught exception to the user. The minimum requirement that the error will be clear if it is an authentication error (wrong credentials) or a network error (wrong host). This is one of the most sensitive parts of setting up an integration and we want to help our users.